### ISSUE TITLE
Deploy key is destroyed and created at each run

### ISSUE THREAD

**Issue by benosman** [terraform-providers/terraform-provider-gitlab#25]
### Terraform Version
0.10.6

### Affected Resource(s)
- gitlab_deploy_key

### Terraform Configuration Files
```hcl
resource "gitlab_deploy_key" "default" {
  project = "group_name/project_name"
  title   = "Project Name"
  key     = "${var.deploy_key}"
}

```
### Expected Behavior
The first run should create the key, and as long as there are no changes either to the key parameter or on gitlab itself, then subsequent runs should just verify the key is the same and skip any action.

### Actual Behavior
The first run creates the key as expected, but subsequent runs always destroy and create the key rather than verifiying that it has not changed.

Looking at the code it seems that's because there is no update handler for this resource.

### Steps to Reproduce
1. `terraform apply`

**Comment by roidelapluie** [comment]
I do not have that behaviour, I will investigate

**Comment by roidelapluie** [comment]
can you please provide logs?

TF_LOG=1 terraform plan

**Comment by benosman** [comment]
I created a simple resource set and it still shows this behaviour

```hcl
# Configure the GitLab Provider
provider "gitlab" {
  token = "${var.token}"
}

resource "tls_private_key" "gitlab" {
  algorithm   = "RSA"
  rsa_bits = "2048"
}

resource "gitlab_deploy_key" "default" {
  project = "${var.project_id}"
  title   = "salt deploy key"
  key     = "${tls_private_key.gitlab.public_key_openssh}"
}

```

Here is the log file (set to TRACE - TF_LOG=1 doesn't work for me on windows)

https://pastebin.com/DFP0P4qv

**Comment by roidelapluie** [comment]
Fix proposed in #34
